main/autogen/autogen.cc

```diff
@@ -301,13 +300,22 @@ class AutogenWalk {
     }
 
     void postTransformSend(core::Context ctx, ast::ExpressionPtr &tree) {
-        auto *original = ast::cast_tree<ast::Send>(tree);
         // if this send was something we were ignoring (i.e. a `keepForIde` or an `include` or `require`) then pop this
-        if (!ignoring.empty() && ignoring.back() == original) {
+        if (!ignoring.empty() && ignoring.back() == tree.getTagged()) {
             ignoring.pop_back();
         }
     }
 
+    void preTransformKeepForIDE(core::Context ctx, ast::ExpressionPtr &tree) {
+        ignoring.push_back(tree.getTagged());
+    }
+
+    void postTransformKeepForIDE(core::Context ctx, ast::ExpressionPtr &tree) {
+        ENFORCE(!ignoring.empty());
+        ENFORCE(ignoring.back() == tree.getTagged());
+        ignoring.pop_back();
+    }
```

Comment:
Taking a step back, do we need to do this at all?

It looks like the old code was only looking at `keepForIde` because it wanted to say that those node types did not define behavior, because normally `Send` nodes are said to define behavior.

Now that we don't have `keepForIde` sends, is it possible to just delete all the code that deals with `KeepForIDE` from this file? e.g. leave `ignoring` as a `vector<Send *>`?

Reply:
Before, we had sends like `Sorbet::Private::Static.keep_for_ide(Some::Constant::Of::Mine)`.  We needed to ignore any constants passed as an argument to that method.

Now, we have a magic `<keep-for-ide(Some::Constant::Of::Mind)>` node.  We still need to ignore the constant that appears as its lone argument...but we also need to keep ignoring constants in other `Send`-related contexts.

So I think we need to record both kinds of nodes here.  It would be cool if we had some way of controlling how the treewalk continues from within the `TreeWalk` methods, but that would be a decent-sized task.

For the record, I don't particularly like this solution in `autogen.cc`; `taggedStorage` being a quasi-public API is gross, but it certainly is convenient to use here, rather than having to manually do pointer tagging ourself with the raw `Send` or `KeepForIDE` pointers.